### DISCUSSION TITLE
mini.tabline: How to filter buffers

### DISCUSSION THREAD

**Discussion by wiseodd** [nvim-mini/mini.nvim#976] [Q&A] [answered]
Hi, thanks for the plugins! Currently I'm customizing `mini.tabline` and one thing I notice is that the LSP quickfix floating (if that's the correct term) window is treated as buffer, and so it's shown in the tabline. 

I checked the helpfile but couldn't find any information on how to filter it out. Could you please tell me a way to do it?

**Comment by echasnovski** [comment, marked as answer]
The 'mini.tabline' module is designed to [show all listed buffers in order of their identifiers](https://github.com/echasnovski/mini.nvim/blob/ebc8cb0c1f73ee6c53e0560957a37a2307a70f98/doc/mini-tabline.txt#L8). Here "listed" buffer is a special term describing buffers with set 'buflisted' option; they are essentially the buffers which "should be listed". Preserving order is important because `:bnext` and `:bprev` (or `[b` and `]b` from 'mini.bracketed') then traverse tabline in order.

That said, recently there is a new [`format` config value](https://github.com/echasnovski/mini.nvim/blob/ebc8cb0c1f73ee6c53e0560957a37a2307a70f98/doc/mini-tabline.txt#L93) which can be used to finalize what text is used to show buffer in a tabline. You can use it to return empty string for buffers which you don't want to be shown in tabline. Here is an example which will not show buffers with 'filetype' equal to `'vim'`:

```lua
require('mini.tabline').setup({
  format = function(buf_id, label)
    if vim.bo[buf_id].filetype == 'vim' then return '' end
    return MiniTabline.default_format(buf_id, label)
  end,
})
```

Unfortunately, I don't really understand what you mean by "LSP quickfix floating", so can not help with exact final condition for filtering out such buffers. But usually it is done by 'filetype' option (possibly together with 'buftype' option).

------

Note, that I'd recommend to *not* filter out buffers based on your criteria, as the default behavior shown to be the most robust approach (as of now).

**Reply by wiseodd** [reply]
Thanks for the prompt reply! The quickfix window I referred to was this: <https://neovim.io/doc/user/quickfix.html>, invoked by `vim.diagnostic.setloclist`. 

Your solution works by filtering `filetype == 'qf'`. Thanks a lot!

```diff
require('mini.tabline').setup({
  format = function(buf_id, label)
+   if vim.bo[buf_id].filetype == 'qf' then return '' end
    return MiniTabline.default_format(buf_id, label)
  end,
})
```

**Reply by echasnovski** [reply]
Ah, I see.

Another approach is to set make such buffers not listed. Here is one Vimscript approach: `au FileType qf setlocal nobuflisted`. Better yet, create a '~/.config/nvim/after/ftplugin/qf.lua' file and put `vim.bo.buflisted = false` there.

**Reply by wiseodd** [reply]
Thanks for the hints! I settled with this:

```lua
vim.api.nvim_create_autocmd('FileType', {
  pattern = { 'qf' },
  callback = function(event)
    vim.bo[event.buf].buflisted = false
  end,
})
```
